src/Symfony/Component/Config/Loader/FileLoader.php

```diff
@@ -59,23 +59,37 @@ public function getLocator()
     /**
      * Imports a resource.
      *
-     * @param mixed       $resource       A Resource
-     * @param string|null $type           The resource type or null if unknown
-     * @param bool        $ignoreErrors   Whether to ignore import errors or not
-     * @param string|null $sourceResource The original resource importing the new resource
+     * @param mixed                $resource       A Resource
+     * @param string|null          $type           The resource type or null if unknown
+     * @param bool                 $ignoreErrors   Whether to ignore import errors or not
+     * @param string|null          $sourceResource The original resource importing the new resource
+     * @param string|string[]|null $exclude        Glob patterns to exclude from the import
      *
      * @return mixed
      *
      * @throws LoaderLoadException
      * @throws FileLoaderImportCircularReferenceException
      * @throws FileLocatorFileNotFoundException
      */
-    public function import($resource, $type = null, $ignoreErrors = false, $sourceResource = null)
+    public function import($resource, $type = null, $ignoreErrors = false, $sourceResource = null/*, $exclude = null*/)
     {
+        if (\func_num_args() < 5 && __CLASS__ !== \get_class($this) && __CLASS__ !== (new \ReflectionMethod($this, __FUNCTION__))->getDeclaringClass()->getName() && !$this instanceof \PHPUnit\Framework\MockObject\MockObject && !$this instanceof \Prophecy\Prophecy\ProphecySubjectInterface) {
+            @trigger_error(sprintf('The "%s()" method will have a new "$exclude = null" argument in version 5.0, not defining it is deprecated since Symfony 4.4.', __METHOD__), E_USER_DEPRECATED);
+        }
+        $exclude = \func_num_args() >= 5 ? func_get_arg(4) : null;
+
         if (\is_string($resource) && \strlen($resource) !== $i = strcspn($resource, '*?{[')) {
+            $excluded = [];
```

Comment:
Is it expected, that `exclude` is only handled when the resource contains `*?{[`?
`exclude` does not have any effect when using a directory as resource, like in default recipe:
https://github.com/symfony/recipes/blob/b364e60751fba4fc5da103304c1f61259d370e10/doctrine/annotations/1.0/config/routes/annotations.yaml#L2

Reply:
Thx @gharlan and @Growiel I stumbled upon this strange behaviour today and thanks to your hints I did not have to spend too much time debugging this :) Did you open an issue for this though?

Reply:
Hello, I guess you should open an issue if you think it's a bug or if this case should be handled

Reply:
It is indeed a bug. Do you need a new issue or handle it here ?

Reply:
Please create a new issue. Comments on already merged pull requests are likely to get lost.